rflx/generator/common.py

```diff
@@ -418,6 +429,16 @@ def context_predicate(
     message: model.Message, composite_fields: Sequence[model.Field], prefix: str
 ) -> Expr:
     def cursors_invariant() -> Expr:
+        """
+        Create the invariant that defines valid representations of fields.
+
+        Each field cursor represents the state of one parsed or serialized message field.
+        This invariant ensures for all structurally valid fields that
+
+            - the field bounds are inside the range of verified buffer part,
+            - the field has a length greater than zero,
```

Comment:
Is that always the case? How are e.g. opaque fields with a length of 0 handled here?

Reply:
Good point, the description was wrong here. I changed it to "the field size is greater or equal to zero".